ui/apps/ui/src/app/components/sort-by-functionality/sort-by-functionality.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { Router } from '@angular/router';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { CustomRoute } from '@collections/services/custom-route.service';
+import { sanitizeQuery } from '@components/search-input/query.sanitizer';
+import { filter, take } from 'rxjs';
+import { DEFAULT_SORT } from '@components/sort-by-functionality/sort-value-default';
+
+@UntilDestroy()
+@Component({
+  selector: 'ess-sort-by-functionality',
+  template: `<div class="col-sm-4 sort_container">
+    <label for="sorts" i18n>Sort By</label>
+    <select
+      [formControl]="selectedSortOptionControl"
+      id="sorts"
+      class="form-select"
+    >
+      <option value="r" i18n>Relevancy</option>
+      <option value="dmr" i18n>Date - Most recent</option>
+      <option value="dlr" i18n>Date – Least recent</option>
+      <option value="mp" i18n>Most popular</option>
+    </select>
+  </div>`,
+  styles: [
+    `
+      :host {
+        display: contents;
+      }
+
+      label {
+        width: 100px;
+        font-family: 'Inter';
+        font-style: normal;
+        font-weight: 400;
+        font-size: 14px;
+        line-height: 20px;
+        color: #4f4f4f;
+      }
+
+      select {
+        font-family: 'Inter';
+        font-style: normal;
+        font-weight: 500;
+        font-size: 14px;
+        line-height: 24px;
+        color: #101828;
+      }
+
+      select:focus {
+        -webkit-box-shadow: none;
+        box-shadow: none;
+        border: 1px solid #ced4da;
+      }
+
+      .sort_container {
+        width: 18rem;
+        display: flex;
+        align-items: center;
+      }
+    `,
+  ],
+})
+export class SortByFunctionalityComponent implements OnInit {
+  public selectedSortOptionControl: FormControl = new FormControl(DEFAULT_SORT);
```

Comment:
FormControl's in Angular 14+ are typed. It would be good to define types elsewhere (for example in `sort-value-default.ts` e.g.:

```typescript
// sort-value-default.ts
export type sortType = 'r' | 'dmr' | 'dlr' | 'mp';
```

```typescript
// sort-by-functionality.component.ts
public selectedSortOptionControl: FormControl<sortType> =
    new FormControl<sortType>(DEFAULT_SORT, { nonNullable: true });
```

Reply:
Nice. fixed.